File: week_1/Day 1/217_contains_duplicate.py
# Given an integer array nums, return true if any value appears at least twice in the array,
#  and return false if every element is distinct.

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        count = {}

        for num in nums:
            if num not in count:
                count[num] = 1
            else:
                count[num] += 1
                return True

        return False

File: week_1/Day 1/js/217_contains_duplicate.js
// # Given an integer array nums, return true if any value appears at least twice in the array,
// #  and return false if every element is distinct.

// #  Javascript

var containsDuplicate = (nums, numsSet = new Set()) => {
    for (const num of nums) {/* Time O(N) */
        if (numsSet.has(num)) return true;

        numsSet.add(num);       /* Space O(N) */
    }

    return false;
};